control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -24,10 +30,20 @@ const (
 	kindGateway = "Gateway"
 )
 
+// GatewayControllerConfig holds the values necessary for configuring the GatewayController
+type GatewayControllerConfig struct {
+	ConsulClientConfig  *consul.Config
+	ConsulServerConnMgr consul.ServerConnectionManager
+	NamespacesEnabled   bool
+	Partition           string
+	Logger              logr.Logger
```

Comment:
I think you could remove the Logger here and just use the Logger that the user has passed in on the `mgr` so both the Controller and the Cache are using the same logger.

Reply:
ohhh good call